src/rdkafka_assignor.c

```diff
@@ -586,10 +599,17 @@ int rd_kafka_assignors_init(rd_kafka_t *rk, char *errstr, size_t errstr_size) {
                 if (!rkas->rkas_enabled) {
                         rkas->rkas_enabled = 1;
                         rk->rk_conf.enabled_assignor_cnt++;
+                        rkas->rkas_index = index;
+                        index++;
                 }
 
                 s = t;
         }
+        rd_list_sort(&rk->rk_conf.partition_assignors, compare);
+
+        /* Clear the SORTED flag because the list is sorted according to the
+         * rkas_index, but will do the search using rkas_protocol_name. */
+        rk->rk_conf.partition_assignors.rl_flags ^= RD_LIST_F_SORTED;
```

Comment:
Clear bits this way:
`rk->rk_conf.partition_assignors.rl_flags &= ~RD_LIST_F_SORTED;`

Reply:
I think we will get the same result with `^=`, is there any reason that we don't want to use `^=` ?

Reply:
XOR will toggle the bit, which makes assumptions that it has been set, but we want to clear it.